test/unit/org/apache/cassandra/config/DataRateSpecTest.java

```diff
@@ -79,37 +99,95 @@ public void testFromSymbol()
+        assertEquals(new DataRateSpec.IntMebibytesPerSecondBound("10240B/s"), new DataRateSpec.IntMebibytesPerSecondBound("10KiB/s"));
+        assertNotEquals(new DataRateSpec.IntMebibytesPerSecondBound("0KiB/s"), new DataRateSpec.IntMebibytesPerSecondBound("10MiB/s"));
+
+        assertEquals(new DataRateSpec.IntMebibytesPerSecondBound("10B/s"), new DataRateSpec.LongBytesPerSecondBound("10B/s"));
+        assertEquals(new DataRateSpec.IntMebibytesPerSecondBound("10KiB/s"), new DataRateSpec.LongBytesPerSecondBound("10240B/s"));
+        assertEquals(new DataRateSpec.IntMebibytesPerSecondBound("10240B/s"), new DataRateSpec.LongBytesPerSecondBound("10KiB/s"));
+        assertNotEquals(new DataRateSpec.IntMebibytesPerSecondBound("0KiB/s"), new DataRateSpec.LongBytesPerSecondBound("10MiB/s"));
+    }
+
+    @Test
+    public void thereAndBackLongBytesPerSecondBound()
+    {
+        Gen<DataRateSpec.DataRateUnit> unitGen = SourceDSL.arbitrary().enumValues(DataRateSpec.DataRateUnit.class);
+        // DataRateSpec is a special case where we have double so we can accomodate the backward compatibility for parameters which were in
+        // megabits per second before without losing precision
+        // Extremely big numbers might be not completely accurate, that is why here Long.MAX_VALUE is not failing for bytes being >= Long.MAX_VALUE
+        Gen<Long> valueGen = SourceDSL.longs().between(0, Long.MAX_VALUE/1024L/1024L); // the biggest value in MiB/s that won't lead to B/s overflow
+        qt().forAll(valueGen, unitGen).check((value, unit) -> {
+            DataRateSpec there = new DataRateSpec.LongBytesPerSecondBound(value, unit);
+            DataRateSpec back = new DataRateSpec.LongBytesPerSecondBound(there.toString());
+            DataRateSpec BACK = new DataRateSpec.LongBytesPerSecondBound(there.toString());
+            return there.equals(back) && there.equals(BACK);
+        });
     }
 
     @Test
-    public void thereAndBack()
+    public void thereAndBackIntMebibytesPerSecondBound()
     {
         Gen<DataRateSpec.DataRateUnit> unitGen = SourceDSL.arbitrary().enumValues(DataRateSpec.DataRateUnit.class);
-        Gen<Long> valueGen = SourceDSL.longs().between(0, Long.MAX_VALUE);
+        Gen<Long> valueGen = SourceDSL.longs().between(0, Integer.MAX_VALUE-1); // max MiB/s
         qt().forAll(valueGen, unitGen).check((value, unit) -> {
-            DataRateSpec there = new DataRateSpec(value, unit);
-            DataRateSpec back = new DataRateSpec(there.toString());
-            DataRateSpec BACK = new DataRateSpec(there.toString());
+            DataRateSpec there = new DataRateSpec.IntMebibytesPerSecondBound(value, unit);
+            DataRateSpec back = new DataRateSpec.IntMebibytesPerSecondBound(there.toString());
+            DataRateSpec BACK = new DataRateSpec.IntMebibytesPerSecondBound(there.toString());
```

Comment:
nitty nit: Should the sequence of names be like... `here` -> `there` -> `back`? :)

(I guess in that case the second `DataRateSpec.IntMebibytesPerSecondBound()` uses `here` and the third keeps using `there`.)

Same for `thereAndBackLongBytesPerSecondBound()` above.

Reply:
I can't recall why the second BACK actually... I suspect some copy paste. If you look also at the other two test classes... It just does one more time from toString. I think it should be only:
```
@Test
    public void thereAndBackLongBytesPerSecondBound()
    {
        Gen<DataRateSpec.DataRateUnit> unitGen = SourceDSL.arbitrary().enumValues(DataRateSpec.DataRateUnit.class);
        // DataRateSpec is a special case where we have double so we can accomodate the backward compatibility for parameters which were in
        // megabits per second before without losing precision
        // Extremely big numbers might be not completely accurate, that is why here Long.MAX_VALUE is not failing for bytes being >= Long.MAX_VALUE
        Gen<Long> valueGen = SourceDSL.longs().between(0, Long.MAX_VALUE/1024L/1024L); // the biggest value in MiB/s that won't lead to B/s overflow
        qt().forAll(valueGen, unitGen).check((value, unit) -> {
            DataRateSpec there = new DataRateSpec.LongBytesPerSecondBound(value, unit);
            DataRateSpec back = new DataRateSpec.LongBytesPerSecondBound(there.toString());
            return there.equals(back) && back.equals(there);
        });
    }

    @Test
    public void thereAndBackIntMebibytesPerSecondBound()
    {
        Gen<DataRateSpec.DataRateUnit> unitGen = SourceDSL.arbitrary().enumValues(DataRateSpec.DataRateUnit.class);
        Gen<Long> valueGen = SourceDSL.longs().between(0, Integer.MAX_VALUE-1); // max MiB/s
        qt().forAll(valueGen, unitGen).check((value, unit) -> {
            DataRateSpec there = new DataRateSpec.IntMebibytesPerSecondBound(value, unit);
            DataRateSpec back = new DataRateSpec.IntMebibytesPerSecondBound(there.toString());
            return there.equals(back) && back.equals(there);
        });
    }
```

Reply:
That works too :)